lib/components/molecules/AtomicCarousel/AtomicCarousel.js

```diff
@@ -0,0 +1,129 @@
+// @flow
+/**
+ *
+ * AtomicCarousel
+ *
+ */
+import React, { useState, useEffect, useCallback } from 'react';
+import styled from 'styled-components';
+
+import styles from './AtomicCarousel.style';
+import type { Props } from './types';
+import List from '../List';
+import ListItem from '../../atoms/ListItem';
+import { ButtonVanilla } from '../../atoms/Button';
+
+// Rendering all dots as button
+const getSliderDots = (slidesCount, currentItem, setItem) => {
+  const specificNode = useCallback(index => setItem(index));
+  const countElements = [];
+
+  for (let i = 1; i <= slidesCount; i += 1) {
+    countElements.push(
+      <ListItem key={i} className="atomic-item">
+        <ButtonVanilla disabled={currentItem === i} onClick={() => specificNode(i)}>
+          {i}
+        </ButtonVanilla>
+      </ListItem>
+    );
+  }
+
+  return countElements;
+};
+
+// Calculating required number of dots
+const getList = (items, setItem, currentItem, perSlideFrame) => {
```

Comment:
Where does slide count come from? Why can't you find this by children.length

Reply:
It is being calculated by using perSlideFrame and children's length.